### PR TITLE
Feature/swed 2002 update badge colors

### PR DESCRIPTION
**Pull request by CrappyCracker**
## Description

- Added a bunch of color combinations for the badge component. The combinations are made by Filippos.
- Updated the component documentation page to showcase more combinations. Also combined the text and numbers examples into one preview with options to toggle between.
- Added a couple of color variables in the global .less file. 

## Motivation and Context

https://payexjira.atlassian.net/browse/SWED-2002

It was requested to have some more color options on the badge component. It was also requested to create a badge by typing the color instead. i.e. `badge-red`. (old: `badge-warning`)
The old versions are still usable so we dont break anything.

## How Has This Been Tested?

Ran all tests, updated badge tests

## Screenshots (if appropriate):
image

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have updated the **CHANGELOG** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

## Review instructions
[Review instructions](../REVIEW_INSTRUCTIONS.md)

### CODE REVIEW

File: src/App/ComponentsDocumentation/components/Badge/index.js [hunk 1]
```diff
@@ -2,28 +2,40 @@ import React from "react";
...
-const BadgeNumber = () => (
-    <>
-        <h2 id="badgenumber">Badges with numbers</h2>
-        <p>Badges are often used with numbers. Use <CodeTags type="secondary" code=".badge-number" /> when a badge suited for use with numbers is desired.</p>
-        <ComponentPreview language="html" showCasePanel codeFigure>
-            <p>Badge <span className="badge badge-default badge-number">1</span></p>
-            <p>Badge <span className="badge badge-inactive badge-number">8</span></p>
-            <p>Badge <span className="badge badge-warning badge-number">100</span></p>
-        </ComponentPreview>
+        <h2>Do's and Dont's</h2>
```
Comment (eskilsen) [inline src/App/ComponentsDocumentation/components/Badge/index.js:20]: Use h3 here
And something is weird when rendering the ' ´s, don't know what it is, but try out another one. I.E &apos;

File: src/App/ComponentsDocumentation/components/Badge/index.js [hunk 2]
```diff
@@ -2,28 +2,40 @@ import React from "react";
 
 import { ComponentPreview, DocContainer } from "@docutils";
 import CodeTags from "@components/CodeTags";
+import {badgeShowcase} from "./constants"
 
 const Overview = () => (
     <>
```
Comment (eskilsen) [inline src/App/ComponentsDocumentation/components/Badge/index.js:8]: add <section>

File: src/App/ComponentsDocumentation/components/Badge/index.js [hunk 3]
```diff
@@ -2,28 +2,40 @@ import React from "react";
 
 import { ComponentPreview, DocContainer } from "@docutils";
 import CodeTags from "@components/CodeTags";
+import {badgeShowcase} from "./constants"
```
Comment (eskilsen) [inline src/App/ComponentsDocumentation/components/Badge/index.js:5]: need spaces between badgeShowcase
and also, ending semi colon;

File: src/App/ComponentsDocumentation/components/Badge/index.js [hunk 4]
```diff
@@ -2,28 +2,40 @@ import React from "react";
...
         <h2 id="overview">Overview</h2>
-        <p>Use class <CodeTags type="secondary" code=".badge" /> along with <CodeTags type="secondary" code=".badge-{color}" /> to display the badge of your choosing.</p>
-        <ComponentPreview language="html" showCasePanel codeFigure>
-            <p>Badge <span className="badge badge-default">Default</span></p>
-            <p>Badge <span className="badge badge-inactive">Inactive</span></p>
-            <p>Badge <span className="badge badge-warning">Warning</span></p>
-        </ComponentPreview>
-    </>
-);
+        <p>Use class <CodeTags type="secondary" code=".badge" /> along with <CodeTags type="secondary" code=".badge-{color}" /> to display the badge of your choosing. Badges are often used with numbers. Use <CodeTags type="secondary" code=".badge-number" /> when a badge suited for use with numbers is desired.</p>
```
Comment (eskilsen) [inline src/App/ComponentsDocumentation/components/Badge/index.js:10]: Weird sentence.. At least the last part
Maybe: Use <CodeTags type="secondary" code=".badge-number" /> when a badge with only numbers is desired.

File: src/App/ComponentsDocumentation/components/Badge/index.js [hunk 5]
```diff
@@ -2,28 +2,40 @@ import React from "react";
...
+        <ComponentPreview language="html" showCasePanel showCasePanelAdvanced={badgeShowcase} codeFigure removeOuterTag/>
+        <h3>Things to consider</h3>
+        <ul className="list list-bullet">
+            <li>A normal badge does not feature interaction, use the button component instead.</li>
+            <li>If you want removable badges considering using the tag component instead.</li>
```
Comment (eskilsen) [inline src/App/ComponentsDocumentation/components/Badge/index.js:15]: consider*

File: src/App/ComponentsDocumentation/components/Badge/index.js [hunk 6]
```diff
@@ -2,28 +2,40 @@ import React from "react";
...
+            </div>
+            <div className="col-lg-6">
+                <div className="slab slab-plain slab-border-error">
+                    <h3>Don't</h3>
+                    <div className="d-flex justify-content-center mt-4 mb-4">
+                        <span className="badge badge-yellow-light">Transactions from 2022-2023</span>
+                    </div>
+                </div>
+            </div>
+        </div>
```
Comment (eskilsen) [inline src/App/ComponentsDocumentation/components/Badge/index.js:38]: Is there a text missing here? 
Form figma: "If possible, try not to use more than three words. Do not use too long labels or whole sentences."

File: src/App/ComponentsDocumentation/components/Badge/constants.js [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+                <p> <span className="badge badge-yellow badge-number">1</span></p> 
+                <p> <span className="badge badge-gray badge-number">2</span></p> 
+                <p> <span className="badge badge-red badge-number">3</span></p> 
+                <p> <span className="badge badge-blue badge-number">4</span></p> 
+                <p> <span className="badge badge-pink badge-number">5</span></p> 
+                <p> <span className="badge badge-green badge-number">6</span></p> 
+                <p> <span className="badge badge-orange badge-number">7</span></p> 
+                <p> <span className="badge badge-turqouise badge-number">8</span></p> 
+                <p> <span className="badge badge-brown badge-number">9</span></p> 
+            </div>
```
Comment (eskilsen) [inline src/App/ComponentsDocumentation/components/Badge/constants.js:17]: Remove whitespace at end of this line

File: src/App/ComponentsDocumentation/components/Badge/constants.js [hunk 8]
```diff
@@ -0,0 +1,66 @@
...
+                            }
+                        ]
+                    }
+                ]
+            },
+            title: "Badge",
+            description: "Use the Badge Component Symbol to draw attention to another interface element or to display a notification. It enhances the component to which it is attached with additional information, disclosed to the user upon interacting with it. "
+        }
+    ]
+}
```
Comment (eskilsen) [inline src/App/ComponentsDocumentation/components/Badge/constants.js:66]: Add new line after line 66
We use prettifier (I think it is called), an extension that fixes shit like this :D

File: src/less/components/badge.less [hunk 9]
```diff
@@ -3,8 +3,41 @@
...
+    orange: @brand-orange;
+    orange-light: @brand-warning-light;
+    green: @brand-green;
+    green-light: @brand-green-light;
+    blue: @brand-blue;
+    blue-light: @brand-blue-light;
+
+    // Should be reworked to variables containing both text color and bg color?
+
+    // Contrast checker -> https://webaim.org/resources/contrastchecker/
```
Comment (eskilsen) [inline src/less/components/badge.less:40]: I guess you have validated the contrasts? 🤓

Comment (CrappyCracker) [reply src/less/components/badge.less:40]: Yes, I just pasted the one i used 😎

File: src/App/ComponentsDocumentation/components/Badge/index.js (hunk 6 above)
Comment (CrappyCracker) [reply src/App/ComponentsDocumentation/components/Badge/index.js:38]: I moved it to the bulletpoints. Made the suggestion to Filippos

**Review by eskilsen — APPROVED** [review]
Great 💯

**Pull request merged**